src/gnn_reco/data/sqlite_dataset.py

```diff
@@ -153,7 +153,7 @@ def _create_graph(self, features, truth):
         for write_dict in [labels_dict, truth_dict]:
             for key, value in write_dict.items():
                 try:
-                    graph[key] = torch.tensor(value)
+                    graph[key] = torch.tensor(value,dtype=torch.float)
```

Comment:
```suggestion
                    graph[key] = torch.tensor(value, dtype=torch.float)
```

Reply:
This will convert e.g. integers and booleans to floats; for the GNN training itself I reckon this is fine, but perhaps there are other cases that may be affected, e.g. if a boolean attribute is checked? What is the benefit for casting all attributes in this way?

Reply:
This change should only affect labels_dict where values are booleans, and in binary classification cases the target you aim for must be 0 or 1. However if the booleans are needed then bool(0) = False or bool(1)=True could be used to retrieve this.